cms/djangoapps/contentstore/views/preview.py

```diff
@@ -133,6 +133,10 @@ def __init__(self, request):
         super(StudioPermissionsService, self).__init__()
         self._request = request
 
+    def set_request(self, request):
+        """ Set the request param """
+        self._request = request
```

Comment:
I don't really like this approach of keeping this service object in the cache but "fixing" the user value when necessary - it's too easy to forget to "fix" the user value, since developers need to remember to do that in any view that loads this type of block, and if one made such an oversight, then there would be no errors raised to indicate the wrong user value may be in use.